### PR TITLE
Feature/block height

### PR DESCRIPTION
**Pull request by mr-rooftop**

### CODE REVIEW

File: test/integration_test.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 # We got an address from our mock api json file.
 # This address will be recognized as valid by our bot and used throughout the tests.
 VALID_ADDRESS = json.load(open('nodeaccounts.json'))[0]['node_address']
+VALID_ADDRESS2 = json.load(open('nodeaccounts.json'))[0]['node_address']
```
Comment (rekt0x) [inline test/integration_test.py:27]: What `VALID_ADDRESS2` for? Seems unused.

File: thornode_bot.py [hunk 2]
```diff
@@ -43,14 +44,15 @@ def start(update, context):
...
+        context.job_queue.run_repeating(node_checks, interval=30, context={
             'chat_id': update.message.chat.id,
             'user_data': context.user_data
         })
         context.user_data['job_started'] = True
 
     text = 'Heil ok sæll! I am your THORNode Bot. 🤖\n' + \
-           'I will notify you about changes of your node\'s *Status*, *Bond* or *Slash Points*!\n'
+           'I will notify you about changes of your node\'s *Status*, *Bond* or *Slash Points*, ' \
+           'and when your *Block Height* got stuck!\n'
```
Comment (rekt0x) [inline thornode_bot.py:55]: `'and when your *Block Height* got stuck!\n'` --> `'and if your *Block Height* gets stuck!'`
I'd also remove the `\n` at the end(not sure  what that is for)

File: thornode_bot.py [hunk 3]
```diff
@@ -222,9 +218,77 @@ def check_thornode(context):
...
+
+    block_height = get_block_height()
+
+    # Check if block height got stuck
+    if 'block_height' in user_data and block_height <= user_data['block_height']:
+
+        # Increase stuck count to know if we already sent a notification
+        block_height_stuck_count = user_data['block_height_stuck_count']
+        block_height_stuck_count = block_height_stuck_count + 1
+        user_data['block_height_stuck_count'] = block_height_stuck_count
```
Comment (rekt0x) [inline thornode_bot.py:242]: Could be a one-liner, something like this:
```
user_data['block_height_stuck_count'] += 1
```
or
```
user_data['block_height_stuck_count'] = user_data['block_height_stuck_count'] + 1
```

File: thornode_bot.py [hunk 4]
```diff
@@ -222,9 +218,77 @@ def check_thornode(context):
...
+               'Block height stuck at: ' + block_height + '\n\n' + \
+               'Please check your Thornode immediately!'
+        context.bot.send_message(chat_id, text)
+
+    # Show buttons if there were changes or block height just got (un)stuck
+    # Stuck count:
+    # 0 == everthings alright
+    # 1 == just got stuck
+    # -1 == just got unstuck
+    # > 1 == still stuck
```
Comment (rekt0x) [inline thornode_bot.py:270]: It would be nice to use an Enum for that!

File: thornode_bot.py [hunk 5]
```diff
@@ -222,9 +218,77 @@ def check_thornode(context):
...
+    with open('test/status.json') as json_read_file:
+        node_data = json.load(json_read_file)
+
+    block_height = node_data['result']['sync_info']['latest_block_height']
+    new_block_height = int(block_height) + 1
+    node_data['result']['sync_info']['latest_block_height'] = str(new_block_height)
+
+    with open('test/status.json', 'w') as json_write_file:
+        json.dump(node_data, json_write_file)
+
```
Comment (rekt0x) [inline thornode_bot.py:291]: Why is this function in `thornode_bot.py`? Belongs into the `/tests` directory in my opinion.

**Comment by rekt0x** [comment]
Overall, nice implementation 👍

File: thornode_bot.py (hunk 4 above)
Comment (mr-rooftop) [reply thornode_bot.py:270]: Well, the current logic works with incrementing the stuck_count. If we use an enum, there need to be a few more IFs. 
Maybe check out the logic, I couldn't find an easier solution to notify the user only once the block height gets stuck, and only once when it's increasing again.

File: thornode_bot.py (hunk 5 above)
Comment (mr-rooftop) [reply thornode_bot.py:291]: Hm.
It's not about testing, it's about running it in DEBUG mode, which means in our case running it in local mode.
Let's move it to testing, but start it as a subprocess from thornode_bot.py if DEBUG==True ok?

File: thornode_bot.py (hunk 4 above)
Comment (rekt0x) [reply thornode_bot.py:268]: Ok!

File: thornode_bot.py (hunk 5 above)
Comment (rekt0x) [reply thornode_bot.py:289]: Sorry, my fault. That's fine!

**Pull request merged**